pkg/resource/template/resource.go

```diff
@@ -155,16 +155,30 @@ func (t *TemplateResource) createStageFile() error {
 	}
 
 	if err = tmpl.Execute(temp, nil); err != nil {
-		temp.Close()
-		os.Remove(temp.Name())
+		// The key error to return is the failure to execute.
+		// to preserve that error ignore the errors in close and clean
+		temp.Close()           // nolint:errcheck
+		os.Remove(temp.Name()) // nolint:errcheck
 		return err
 	}
-	defer temp.Close()
+	defer func() {
+		if ferr != nil {
+			log.Error(ferr.Error())
+		}
+		if e := temp.Close(); e != nil {
+			// Just log the error but don't carry it up the calling stack.
+			log.Error(e.Error())
+		}
+	}()
 
 	// Set the owner, group, and mode on the stage file now to make it easier to
 	// compare against the destination configuration file later.
-	os.Chmod(temp.Name(), t.FileMode)
-	os.Chown(temp.Name(), t.Uid, t.Gid)
+	if err := os.Chmod(temp.Name(), t.FileMode); err != nil {
+		return err
```

Comment:
If you log the error here in these branches, then you don't need to change the return error to be parameterized.